src/sympc/module/nn/functional.py

```diff
@@ -37,3 +49,327 @@ def mse_loss(pred: MPCTensor, target: MPCTensor, reduction: str = "mean") -> MPC
+    will run the max_pool2d kernel.
+
+    Args:
+        x (ShareTensor): the ShareTensor on which to apply the reshaping
+        kernel_size (Tuple[int, int]): the kernel size
+        stride (Tuple[int, int]): the stride size
+        padding (Tuple[int, int]): the padding size
+        dilation (Tuple[int, int]): the dilation size
+
+    Returns:
+        The prepared share tensor (reshaped)
+    """
+    session = get_session(x.session_uuid)
+    tensor = x.tensor.numpy()
+
+    padding = [(0, 0)] * len(tensor.shape[:-2]) + [
+        (padding[0], padding[0]),
+        (padding[1], padding[1]),
+    ]
+    tensor_type = session.tensor_type
+
+    padding_value = 0
+    if session.rank == 0:
+        # ATTENTION: Min value for max_pool2d that works -25
+        padding_value = -25
+
+    padding_value = x.fp_encoder.encode(padding_value)
+    tensor = np.pad(tensor, padding, mode="constant", constant_values=padding_value)
+
+    output_shape = tensor.shape[:-2]
+    output_shape += (
+        (tensor.shape[-2] - kernel_size[0]) // stride[0] + 1,
+        (tensor.shape[-1] - kernel_size[1]) // stride[1] + 1,
+    )
+    output_shape += kernel_size
+
+    output_strides = tensor.strides[:-2]
+    output_strides += (stride[0] * tensor.strides[-2], stride[1] * tensor.strides[-1])
+    output_strides += tensor.strides[-2:]
+
+    window_view_share = torch.tensor(
+        np.lib.stride_tricks.as_strided(
+            tensor, shape=output_shape, strides=output_strides
+        ),
+        dtype=tensor_type,
+    )
+
+    window_view_share = window_view_share.reshape(-1, *kernel_size)
+
+    res_share = ShareTensor(config=x.config)
+    res_share.tensor = window_view_share
+    return res_share
+
+
+def max_pool2d(
+    x: MPCTensor,
+    kernel_size: Union[int, Tuple[int, int]],
+    stride: Optional[Union[int, Tuple[int, int]]] = None,
+    padding: Union[int, Tuple[int, int]] = 0,
+    dilation: Union[int, Tuple[int, int]] = 1,
+) -> MPCTensor:
+    """Compute the max pool for a tensor with 2 dimension.
+
+    Args:
+        x (MPCTensor): the MPCTensor on which to apply the operation
+        kernel_size (Union[int, Tuple[int, int]]): the kernel size
+            in case it is passed as an integer then that specific value is used for height and width
+        stride (Union[int, Tuple[int, int]]): the stride size
+            in case it is passed as an integer then that specific value is used for height and width
+        padding (Union[int, Tuple[int, int]]): the padding size
+            in case it is passed as an integer then that specific value is used for height and width
+        dilation (Union[int, Tuple[int, int]]): the dilation size
+            in case it is passed as an integer then that specific value is used for height and width
+
+    Returns:
+        A tuple representing maximum values and the indices (as a one hot encoding
+    """
+    kernel_size, stride, padding, dilation = _sanity_check_max_pool2d(
```

Comment:
I think we need a check to see if the input x is bigger than kernel size. Sometimes this error does occur unintentionally when designing networks. Also, add a test for it. 

Or did I miss this check somewhere.

Reply:
The way PyTorch handles it. 
https://discuss.pytorch.org/t/kernel-size-cant-greater-than-actual-input-size/21579